File: discord_bot.js

// Discord bot using Discord.js
const { Client, GatewayIntentBits, Events } = require('discord.js');
const dotenv = require('dotenv');

// Load environment variables
dotenv.config();

// Create a new client instance with necessary intents
const client = new Client({
  intents: [
    GatewayIntentBits.Guilds,
    GatewayIntentBits.GuildMessages,
    GatewayIntentBits.MessageContent,
  ]
});

// When the client is ready, run this code (only once)
client.once(Events.ClientReady, () => {
  console.log(`${client.user.tag} has connected to Discord!`);
  console.log(`Bot is in ${client.guilds.cache.size} servers`);
});

// Listen for messages
client.on(Events.MessageCreate, async message => {
  // Ignore messages from bots
  if (message.author.bot) return;
  
  // Command handling
  if (message.content.startsWith('!hello')) {
    await message.reply(`Hello ${message.author.username}!`);
  }
  
  if (message.content.startsWith('!ping')) {
    const sent = await message.reply('Pinging...');
    sent.edit(`Pong! Latency: ${Date.now() - message.createdTimestamp}ms`);
  }
});

// Get token from environment variables
const TOKEN = process.env.DISCORD_TOKEN;

if (!TOKEN) {
  console.warn("Warning: No DISCORD_TOKEN found in environment variables!");
  console.warn("Please add your bot token in the Secrets tab in Replit");
  console.warn("You need to create a bot at https://discord.com/developers/applications");
} else {
  // Login to Discord
  client.login(TOKEN);
}

File: discord_bot.py

import os
import discord
from discord.ext import commands

# Set up the bot with command prefix '!'
intents = discord.Intents.default()
intents.message_content = True
bot = commands.Bot(command_prefix='!', intents=intents)

@bot.event
async def on_ready():
    print(f'{bot.user.name} has connected to Discord!')
    print(f'Bot is in {len(bot.guilds)} servers')

@bot.command(name='hello')
async def hello(ctx):
    """Sends a hello message"""
    await ctx.send(f'Hello {ctx.author.name}!')

@bot.command(name='ping')
async def ping(ctx):
    """Check the bot's latency"""
    await ctx.send(f'Pong! Latency: {round(bot.latency * 1000)}ms')

# Replace 'YOUR_BOT_TOKEN' with your actual Discord bot token
# You should store this in Replit's Secrets tab for security
TOKEN = os.getenv('DISCORD_TOKEN')

if not TOKEN:
    print("Warning: No DISCORD_TOKEN found in environment variables!")
    print("Please add your bot token in the Secrets tab in Replit")
    print("You need to create a bot at https://discord.com/developers/applications")
else:
    bot.run(TOKEN)
